src/tests/application/verifier_tests.cpp

```diff
@@ -0,0 +1,43 @@
+#include <gtest/gtest.h>
+#include "AbstractDatabaseInterfaceMock.h"
+#include "../../application/EntryRepository.h"
+#include "../../application/PasswordVerifier.h"
+#include "../../application/InstanceManager.h"
+#include "../../plugins/encryption/DefaultHash.h"
+#include "../../application/Hash.h"
+#include <stdexcept>
+
+using namespace DDD;
+
+ACTION(ThrowDomainError) {
+    throw std::domain_error("");
+}
+
+TEST(SetMasterPassword, DDDTests) {
+    AbstractDatabaseInterfaceMock mock;
+
+    Hash* hash = new DefaultHash();
```

Comment:
Why is this allocated on the heap?

Reply:
because stack objects don't support polymorphic function calls, as they have no vtable

Reply:
Ah that makes sense, I didn't know that :+1: